<ui version="4.0" >
 <class>UIXUPProjectEditor</class>
 <widget class="QDialog" name="UIXUPProjectEditor" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>510</width>
    <height>480</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>XUP Project Editor : %1</string>
  </property>
  <layout class="QVBoxLayout" >
   <item>
    <layout class="QHBoxLayout" >
     <item>
      <widget class="QLabel" name="lScope" >
       <property name="text" >
        <string>Scope :</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="tbAddScope" >
       <property name="icon" >
        <iconset resource="../resources/XUPManager.qrc" >:/actions/add.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="tbRemoveScope" >
       <property name="icon" >
        <iconset resource="../resources/XUPManager.qrc" >:/actions/remove.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="pTreeComboBox" name="cbScope" >
       <property name="sizePolicy" >
        <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lOperator" >
       <property name="text" >
        <string>Operator :</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cbOperator" />
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" >
     <item>
      <widget class="QLabel" name="label" >
       <property name="sizePolicy" >
        <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text" >
        <string>Property :</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cbPages" >
       <item>
        <property name="text" >
         <string>Project Files</string>
        </property>
        <property name="icon" >
         <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/project_files.png</iconset>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Others Variables</string>
        </property>
        <property name="icon" >
         <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/others_variables.png</iconset>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Builder</string>
        </property>
        <property name="icon" >
         <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/builder.png</iconset>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Compiler</string>
        </property>
        <property name="icon" >
         <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/compiler.png</iconset>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Debugger</string>
        </property>
        <property name="icon" >
         <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/debugger.png</iconset>
        </property>
       </item>
       <item>
        <property name="text" >
         <string>Interpreter</string>
        </property>
        <property name="icon" >
         <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/interpreter.png</iconset>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="swPages" >
     <property name="currentIndex" >
      <number>1</number>
     </property>
     <widget class="QWidget" name="wProjectFiles" >
      <layout class="QGridLayout" >
       <property name="leftMargin" >
        <number>0</number>
       </property>
       <property name="topMargin" >
        <number>0</number>
       </property>
       <property name="rightMargin" >
        <number>0</number>
       </property>
       <property name="bottomMargin" >
        <number>0</number>
       </property>
       <item rowspan="4" row="0" column="0" >
        <widget class="QTreeView" name="tvProjectFiles" >
         <property name="editTriggers" >
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1" >
        <widget class="QPushButton" name="pbAddProjectFiles" >
         <property name="text" >
          <string>Add Files...</string>
         </property>
         <property name="icon" >
          <iconset resource="../resources/XUPManager.qrc" >:/actions/add.png</iconset>
         </property>
        </widget>
       </item>
       <item row="1" column="1" >
        <widget class="QPushButton" name="pbRemoveProjectFile" >
         <property name="text" >
          <string>Remove File...</string>
         </property>
         <property name="icon" >
          <iconset resource="../resources/XUPManager.qrc" >:/actions/remove.png</iconset>
         </property>
        </widget>
       </item>
       <item row="2" column="1" >
        <widget class="QPushButton" name="pbModifyProjectFile" >
         <property name="text" >
          <string>Modify File...</string>
         </property>
         <property name="icon" >
          <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/modify.png</iconset>
         </property>
        </widget>
       </item>
       <item row="3" column="1" >
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" >
          <size>
           <width>142</width>
           <height>241</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="wOthersVariables" >
      <layout class="QHBoxLayout" >
       <property name="leftMargin" >
        <number>0</number>
       </property>
       <property name="topMargin" >
        <number>0</number>
       </property>
       <property name="rightMargin" >
        <number>0</number>
       </property>
       <property name="bottomMargin" >
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="gbOthersVariables" >
         <property name="title" >
          <string>Variables</string>
         </property>
         <layout class="QGridLayout" >
          <item row="0" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1" >
           <widget class="QToolButton" name="tbOthersVariablesAdd" >
            <property name="text" >
             <string>Add</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/actions/add.png</iconset>
            </property>
           </widget>
          </item>
          <item row="0" column="2" >
           <widget class="QToolButton" name="tbOthersVariablesEdit" >
            <property name="text" >
             <string>Edit</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/modify.png</iconset>
            </property>
            <property name="popupMode" >
             <enum>QToolButton::MenuButtonPopup</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="3" >
           <widget class="QToolButton" name="tbOthersVariablesRemove" >
            <property name="text" >
             <string>Remove</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/actions/remove.png</iconset>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="4" >
           <widget class="QListView" name="lvOthersVariables" >
            <property name="editTriggers" >
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="selectionBehavior" >
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="gbOthersValues" >
         <property name="title" >
          <string>Values</string>
         </property>
         <layout class="QGridLayout" >
          <item row="0" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>181</width>
              <height>24</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="1" >
           <widget class="QToolButton" name="tbOthersValuesAdd" >
            <property name="text" >
             <string>Add</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/actions/add.png</iconset>
            </property>
            <property name="popupMode" >
             <enum>QToolButton::MenuButtonPopup</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="2" >
           <widget class="QToolButton" name="tbOthersValuesEdit" >
            <property name="text" >
             <string>Edit</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/modify.png</iconset>
            </property>
            <property name="popupMode" >
             <enum>QToolButton::MenuButtonPopup</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="3" >
           <widget class="QToolButton" name="tbOthersValuesRemove" >
            <property name="text" >
             <string>Remove</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/actions/remove.png</iconset>
            </property>
           </widget>
          </item>
          <item row="0" column="4" >
           <widget class="QToolButton" name="tbOthersValuesClear" >
            <property name="text" >
             <string>Clear</string>
            </property>
            <property name="icon" >
             <iconset resource="../resources/XUPManager.qrc" >:/editor/icons/editor/clear.png</iconset>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="5" >
           <widget class="QListView" name="lvOthersValues" >
            <property name="editTriggers" >
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
            <property name="selectionBehavior" >
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="wBuilders" >
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QGroupBox" name="gbBuilders" >
         <property name="title" >
          <string>Use Custom Builder</string>
         </property>
         <property name="checkable" >
          <bool>true</bool>
         </property>
         <property name="checked" >
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" >
          <item row="0" column="0" >
           <widget class="QLabel" name="lBuilders" >
            <property name="text" >
             <string>Builders Available :</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QComboBox" name="cbBuilders" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2" >
           <widget class="QStackedWidget" name="swBuilders" >
            <property name="currentIndex" >
             <number>-1</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="wCompilers" >
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QGroupBox" name="gbCompilers" >
         <property name="title" >
          <string>Use Custom Compiler</string>
         </property>
         <property name="checkable" >
          <bool>true</bool>
         </property>
         <property name="checked" >
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" >
          <item row="0" column="0" >
           <widget class="QLabel" name="lCompilers" >
            <property name="text" >
             <string>Compilers Available :</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QComboBox" name="cbCompilers" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2" >
           <widget class="QStackedWidget" name="swCompilers" >
            <property name="currentIndex" >
             <number>-1</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="wDebuggers" >
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QGroupBox" name="gbDebuggers" >
         <property name="title" >
          <string>Use Custom Debugger</string>
         </property>
         <property name="checkable" >
          <bool>true</bool>
         </property>
         <property name="checked" >
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" >
          <item row="0" column="0" >
           <widget class="QLabel" name="lDebuggers" >
            <property name="text" >
             <string>Debuggers Available :</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QComboBox" name="cbDebuggers" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2" >
           <widget class="QStackedWidget" name="swDebuggers" >
            <property name="currentIndex" >
             <number>-1</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="wInterpreters" >
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QGroupBox" name="gbInterpreters" >
         <property name="title" >
          <string>Use Custom Interpreter</string>
         </property>
         <property name="checkable" >
          <bool>true</bool>
         </property>
         <property name="checked" >
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" >
          <item row="0" column="0" >
           <widget class="QLabel" name="lInterpreters" >
            <property name="text" >
             <string>Interpreters Available :</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QComboBox" name="cbInterpreters" >
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2" >
           <widget class="QStackedWidget" name="swInterpreters" >
            <property name="currentIndex" >
             <number>-1</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="dbbButtons" >
     <property name="standardButtons" >
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Help|QDialogButtonBox::NoButton|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>pTreeComboBox</class>
   <extends>QComboBox</extends>
   <header location="global" >fresh.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>cbPages</tabstop>
  <tabstop>tbAddScope</tabstop>
  <tabstop>tbRemoveScope</tabstop>
  <tabstop>cbScope</tabstop>
  <tabstop>tvProjectFiles</tabstop>
  <tabstop>pbAddProjectFiles</tabstop>
  <tabstop>pbRemoveProjectFile</tabstop>
  <tabstop>pbModifyProjectFile</tabstop>
  <tabstop>tbOthersVariablesAdd</tabstop>
  <tabstop>tbOthersVariablesEdit</tabstop>
  <tabstop>tbOthersVariablesRemove</tabstop>
  <tabstop>lvOthersVariables</tabstop>
  <tabstop>tbOthersValuesAdd</tabstop>
  <tabstop>tbOthersValuesEdit</tabstop>
  <tabstop>tbOthersValuesRemove</tabstop>
  <tabstop>tbOthersValuesClear</tabstop>
  <tabstop>lvOthersValues</tabstop>
  <tabstop>dbbButtons</tabstop>
 </tabstops>
 <resources>
  <include location="../resources/XUPManager.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>cbPages</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>swPages</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>383</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel" >
     <x>309</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dbbButtons</sender>
   <signal>accepted()</signal>
   <receiver>UIXUPProjectEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>516</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel" >
     <x>500</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dbbButtons</sender>
   <signal>rejected()</signal>
   <receiver>UIXUPProjectEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>404</x>
     <y>458</y>
    </hint>
    <hint type="destinationlabel" >
     <x>404</x>
     <y>534</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbBuilders</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>swBuilders</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>103</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel" >
     <x>103</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbInterpreters</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>swInterpreters</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>103</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel" >
     <x>103</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbDebuggers</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>swDebuggers</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>103</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel" >
     <x>103</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbCompilers</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>swCompilers</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>103</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel" >
     <x>103</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
